build/build-dotnet.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -ex
+
+VERSION=$1
+if echo ${VERSION} | grep 'trunk'; then
+    BRANCH=main
+else
+    BRANCH=release/${VERSION}
```

Comment:
The `release/6.0` branch seems to be in active development while the official release is tagged as `v6.0.0` - it's better to use the tags here and only have trunk/main as an active development version.